contrib/parseq-restli-client/src/main/java/com/linkedin/restli/client/ParSeqRestClient.java

```diff
@@ -194,13 +207,56 @@ static String generateTaskName(final Request<?> request) {
     }
   }
 
+  private <T> Task<Response<T>> withD2Timeout(final Task<Response<T>> task, ConfigValue<Long> timeout) {
+    if (timeout.getValue() != null && timeout.getValue() > 0) {
+      // transform TimeoutException to be compatible with current Task timeout exception message
+      String desc = timeout.getSource().map(src -> " src: " + src).orElse("");
+      String timeoutTaskName = "withTimeout " + Math.min(timeout.getValue(), Integer.MAX_VALUE)
```

Comment:
What is a purpose of Math.min()? Is D2 timeout expect Integer type?

Reply:
Yes. In rest.li code, we have made assumption that request timeout should be specified as Integer, such as https://github.com/linkedin/rest.li/blob/master/r2-netty/src/main/java/com/linkedin/r2/transport/http/client/HttpClientFactory.java#L1067, https://github.com/linkedin/rest.li/blob/master/d2/src/main/java/com/linkedin/d2/balancer/clients/RequestTimeoutClient.java#L143, where we directly invoke Number.intValue to get the timeout set in RequestContext. If we don't use Math.min() here, we will get Overflow in RequestTimeoutClient D2 code.